### PR TITLE
Document fields of PushOptions

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by ararslan — APPROVED** [review]

File: base/libgit2/types.jl [hunk 1]
```diff
@@ -393,6 +393,18 @@ end
...
+  * `version`: version of the struct in use, in case this changes later. For now, always `1`.
+  * `parallelism`: if a pack file must be created, this variable sets the number of worker
+     threads which will be spawned by the packbuilder. If `0`, the packbuilder will auto-set
+     the number of threads to use. The default is `1`.
+  * `callbacks`: the callbacks (e.g. for authentication with the remote) to use for the push.
+  * `proxy_opts`: only relevant if the LibGit2 version is greater than or equal to `0.25.0`.
+     Sets options for using a proxy to communicate with a remote. See [`ProxyOptions`](@ref)
+     for more information.
+  * `custom_headers`: only relevant if the LibGit2 version is greater than or equal to `0.24.0`.
+     Extra headers needed for the push operation.
```
Comment (tkelman) [inline base/libgit2/types.jl]: maybe link to libgit2 docs for what this means?

Comment (kshyatt) [inline base/libgit2/types.jl]: In the docs for this function they don't elaborate any more than this.

Comment (kshyatt) [inline base/libgit2/types.jl]: Is that ok, or should i go on a hunt to flesh this out?

Comment (tkelman) [inline base/libgit2/types.jl]: we should maybe be documenting what type each of the fields are for struct docs like this, but sounds like this is obscure enough that there's no common use case for it

**Pull request merged**